bluebutton-data-pipeline-rif-extract/src/main/java/gov/hhs/cms/bluebutton/datapipeline/rif/extract/s3/task/ManifestEntryDownloadTask.java

```diff
@@ -85,6 +101,35 @@ public ManifestEntryDownloadResult call() throws Exception {
 		}
 	}
 
+	/**
+	 * Calculates and returns a Base64 encoded MD5chksum value for the file just
+	 * downloaded from S3
+	 * 
+	 * @param downloadedS3File
+	 *            the {@link InputStream} of the file just downloaded from S3
+	 * @return Base64 encoded md5 value
+	 */
+	public static String computeMD5ChkSum(InputStream downloadedS3File) throws IOException, NoSuchAlgorithmException {
+		// Create byte array to read data in chunks
+		byte[] byteArray = new byte[1024];
+		int bytesCount = 0;
+		MessageDigest md5Digest = MessageDigest.getInstance("MD5");
+
+		// Read file data and update in message digest
+		while ((bytesCount = downloadedS3File.read(byteArray)) != -1) {
+			md5Digest.update(byteArray, 0, bytesCount);
+		}
+
+		// close the stream
+		downloadedS3File.close();
+
+		// Get the hash's bytes
+		byte[] bytes = md5Digest.digest();
+
+		// return complete hash
+		return Base64.getEncoder().encodeToString(bytes);
+	}
```

Comment:
Can you add some metric/timer tracking to this? I'd be curious to see how much time in our runs is spent computing checksums. Here's an example of us tracking how long is spent downloading: https://github.com/CMSgov/bluebutton-data-pipeline/blob/8bfd730847d2ec356c2d7755492928e6a1c286c1/bluebutton-data-pipeline-rif-extract/src/main/java/gov/hhs/cms/bluebutton/datapipeline/rif/extract/s3/task/ManifestEntryDownloadTask.java#L68